website/conferences/views.py

```diff
@@ -181,14 +185,126 @@ def _render_conference_node(node, idx, conf):
         'tags': ' '.join(tags)
     }
 
+def filter_and_sort_conference_data(nodes, conf):
+    """
+    Filter and sort conference submissions
+    Returns: filtered/sorted node queryset.  Sorts on download count by default.
+
+    :param obj nodes: Node queryset - conference submissions
+    :param conf obj: Conference
+    """
+    q = request.args.get('q', '')
+    # Give "sort" a default value, since pagination may be inconsistent with unordered objects.
+    sort = request.args.get('sort', '-downloads')
+
+    if q:
+        format_q = '%' + q + '%'
+        # TODO replace this raw sql with a regular django query when django-include
+        # limitations are fixed: Subqueries can only return one column, but a first visible
+        # contributor subquery also fetches guids, despite .include(None)
+        # This subquery looks for "q" query param in the node title and the first visible contrib's fullname.
+        raw_queryset = nodes.raw(
+            """
+            SELECT *
+            FROM "osf_abstractnode"
+            WHERE (UPPER("osf_abstractnode"."title"::text) LIKE UPPER(%s)
+               OR UPPER(
+                   (SELECT U0."fullname"
+                    FROM "osf_osfuser" U0
+                    INNER JOIN "osf_contributor" U1 ON (U0."id" = U1."user_id")
+                    WHERE (U1."node_id" = ("osf_abstractnode"."id")
+                        AND U1."visible" = true)
+                    ORDER BY U1."_order" ASC
+                    LIMIT 1)::text) LIKE UPPER(%s))
+            """, [format_q, format_q]
+        )
+        # Turns the raw queryset back into a queryset - we still need to sort and paginate it.
+        nodes = nodes.filter(id__in=[node.id for node in raw_queryset])
+
+    if 'title' in sort or 'created' in sort:
+        nodes = nodes.order_by(sort)
+    elif 'author' in sort:
+        nodes = nodes.extra(select={
+            'author': '\
+                (SELECT U0."family_name" \
+                 FROM "osf_osfuser" U0 \
+                 INNER JOIN "osf_contributor" U1 ON (U0."id" = U1."user_id") \
+                 WHERE (U1."node_id" = ("osf_abstractnode"."id") AND U1."visible" = true) \
+                 ORDER BY U1."_order" ASC LIMIT 1)'
+        })
+        nodes = nodes.extra(order_by=[sort])
+    elif 'category' in sort:
+        category_1 = conf.field_names['submission1'] or 'poster'
```

Comment:
I think this and the next line will fail if those keys aren't there, so you might wanna use `conf.field_names.get('submission`, 'poster')` instead